Update to Kubernetes 1.11

linki opened the pull request:
Let's see what the e2es have to say about this.

Changes:
* Update Hyperkube to 1.11.x
* Update CoreOS to the latest stable (as of July 4)
* Remove PodPriority feature flag as it's enabled by default
* ~~Switch from KubeProxy's `iptables` mode to `ipvs` mode~~
* ~~Disable docker version change, therefore it uses 18.03.x with the AMI above which is **not** officially supported by Kubernetes 1.11 (let's see where we end up; 1.12 and 17.03.x work fine and are supported)~~
* Remove `PersistentVolumeClaimResize` feature flag ([enabled by default](https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/?utm_source=cncf&utm_medium=email&utm_campaign=cloudnativemonthly&utm_content=july18))
* Remove `CustomResourceSubresources` feature gate ([enabled by default](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#subresources)).

Commit:
chore: update hyperkube to 1.11.0

linki commented on cluster/manifests/10-vertical-pod-autoscaler/vpa-crd.yaml:
```diff
@@ -27,25 +27,26 @@ spec:
-                        - "On"
-                        - "Off"
+                          - "Auto"
+                          - "Off"
```
This file change is just for testing. Don't merge me.

Commit:
chore: remove obsolete PodPriority feature flag

Commit:
feat: switch kube-proxy from iptables to ipvs

hjacobs commented:
@linki what is the impact of switching to ipvs? We should have a good understanding of how it works and what the benefits/drawbacks are.

Commit:
feat: use AMI's default docker version

linki commented:
@hjacobs It claims to be more performant and seems to become the default option going forward. From [the release notes](https://kubernetes.io/blog/2018/06/27/kubernetes-1.11-release-announcement/): "This change delivers better network throughput, better programming latency, and higher scalability limits for the cluster-wide distributed load-balancer that comprises the Kubernetes Service model"

However, there's [a known-issue](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.11.md#known-issues) in Kubernetes 1.11 so I think we should wait for the next release.

I was just curious if e2e would pass: it does.

Commit:
chore: update CoreOS to the latest version

szuecs commented:
@linki maybe we want not to switch to ipvs https://github.com/kubernetes/kubernetes/issues/57841.
I was suggesting this to do but maybe we should rethink and review what bugs exists on ipvs and which in iptables.

linki commented:
@szuecs yes, it's mentioned as a known issue in the release notes as well. We should wait.

mikkeloscar commented:
Yes, this sounds like quite a serious problem :)

mikkeloscar committed:
Merge branch 'dev' into kube-1.11

Signed-off-by: Mikkel Oscar Lyderik Larsen <m@moscar.net>

# Conflicts:
#	cluster/manifests/kube-proxy/daemonset.yaml
#	cluster/node-pools/master-default/userdata.clc.yaml
#	cluster/node-pools/worker-default/userdata.clc.yaml

mikkeloscar committed:
Remove feature gates enabled by default in v1.11

* `ExpandPersistentVolumes` https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/?utm_source=cncf&utm_medium=email&utm_campaign=cloudnativemonthly&utm_content=july18
* `CustomResourceSubresources` https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#subresources

Signed-off-by: Mikkel Oscar Lyderik Larsen <m@moscar.net>

mikkeloscar committed:
Merge branch 'dev' into kube-1.11

mikkeloscar committed:
Update to Kubernetes v1.11.1

Signed-off-by: Mikkel Oscar Lyderik Larsen <m@moscar.net>

njuettner committed:
Merge branch 'dev' into kube-1.11-patch

njuettner committed:
Patch for Kubernetes v1.11.2

njuettner committed:
Bumping CoreDNS and Heapster to the latest release

linki committed:
Merge pull request #1299 from zalando-incubator/kube-1.11-patch

Kube 1.11 patch

linki commented on cluster/node-pools/master-default/userdata.clc.yaml:
```diff
@@ -868,14 +868,6 @@ storage:
-    mode: 0644
-    contents:
-      inline: yes
-  {{end}}
```
We should fix the docker version again. I would like to try out docker 17 as 18 isn't officially supported.

Furthermore the new (better) way of defining versions for those bundled binaries is using `torcx`: https://coreos.com/blog/toward-docker-17-in-container-linux

aermakov-zalando committed:
Update CA to v1.3.1

njuettner committed:
Cluster Autoscaler v1.3.1, incl. docker 1.12 again

szuecs committed:
Merge pull request #1301 from zalando-incubator/update-autoscaler-1.11

ClusterAutoscaler update

aermakov-zalando committed:
Increase memory reservation for autoscaling buffer pods to 2Gi

aermakov-zalando committed:
ZMON: use a user-defined priority class

njuettner committed:
Merge pull request #1310 from zalando-incubator/fix-zmon

ZMON: use a user-defined priority class

njuettner committed:
test cluster has only one master, one cluster-autoscaler is enough

njuettner committed:
Merge branch 'dev' into kube-1.11

hjacobs committed:
Merge branch 'dev' into kube-1.11

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
Merge pull request #1326 from zalando-incubator/dev-to-kube-1.11

dev-to-kube-1.11

Commit:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Arjun Naik <arjun.naik@zalando.de>

arjunrn committed:
Merge pull request #1331 from zalando-incubator/dev-to-kube-1.11

dev-to-kube-1.11

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.11

aermakov-zalando committed:
Merge pull request #1337 from zalando-incubator/dev-to-kube-1.11

Update 1.11 branch

aermakov-zalando committed:
Run cAdvisor as a daemonset

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1339 from zalando-incubator/cadvisor-1.11

Run cAdvisor as a daemonset

aermakov-zalando committed:
cadvisor: add limits

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

mikkeloscar committed:
Merge pull request #1340 from zalando-incubator/cadvisor-1.11

cadvisor: add limits

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

hjacobs committed:
Merge pull request #1347 from zalando-incubator/dev-to-kube-1.11

Dev to kube 1.11

mikkeloscar committed:
Remove deprecated --allow-privileged kubelet flag

Remove the `--allow-privileged` flag from the kubelet since it now
defaults to `true` and has been marked deprecated and will be removed in
v1.13.

https://github.com/kubernetes/kubernetes/pull/63442

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

aermakov-zalando committed:
Merge pull request #1354 from zalando-incubator/rm-allow-privileged-flag

Remove deprecated --allow-privileged kubelet flag

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

mikkeloscar committed:
Merge pull request #1357 from zalando-incubator/dev-to-kube-1.11

Dev to kube 1.11

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Merge pull request #1361 from zalando-incubator/dev-to-kube-1.11

Dev to kube 1.11

aermakov-zalando committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

hjacobs committed:
Merge pull request #1372 from zalando-incubator/dev-to-kube-1.11

Dev to kube 1.11

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
Merge pull request #1375 from zalando-incubator/dev-to-kube-1.11

dev-to-kube-1.11

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
Merge pull request #1378 from zalando-incubator/dev-to-kube-1.11

dev-to-kube-1.11

szuecs committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

szuecs committed:
Merge pull request #1381 from zalando-incubator/dev-to-kube-1.11

dev-to-kube-1.11

aermakov-zalando committed:
Update to v1.11.3

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

aermakov-zalando committed:
Use the latest Docker version by default

Signed-off-by: Alexey Ermakov <alexey.ermakov@zalando.de>

szuecs committed:
Merge pull request #1384 from zalando-incubator/upstream_docker_default

Use the latest Docker version by default

szuecs committed:
Merge pull request #1383 from zalando-incubator/v1.11.3

Update to v1.11.3

mikkeloscar committed:
Merge branch 'dev' into dev-to-kube-1.11

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

szuecs committed:
Merge pull request #1389 from zalando-incubator/dev-to-kube-1.11

dev-to-kube-1.11

aermakov-zalando commented:
👍

mikkeloscar commented:
:+1:
